src/analyze-action.ts

```diff
@@ -74,22 +74,24 @@ async function sendStatusReport(
     error?.message,
     error?.stack,
   );
-  const report: FinishStatusReport = {
-    ...statusReportBase,
-    ...(stats || {}),
-    ...(dbCreationTimings || {}),
-  };
-  if (config && didUploadTrapCaches) {
-    const trapCacheUploadStatusReport: FinishWithTrapUploadStatusReport = {
-      ...report,
-      trap_cache_upload_duration_ms: Math.round(trapCacheUploadTime || 0),
-      trap_cache_upload_size_bytes: Math.round(
-        await getTotalCacheSize(config.trapCaches, logger),
-      ),
+  if (statusReportBase !== undefined) {
+    const report: FinishStatusReport = {
+      ...statusReportBase,
+      ...(stats || {}),
+      ...(dbCreationTimings || {}),
     };
-    await statusReport.sendStatusReport(trapCacheUploadStatusReport);
-  } else {
-    await statusReport.sendStatusReport(report);
+    if (config && didUploadTrapCaches) {
+      const trapCacheUploadStatusReport: FinishWithTrapUploadStatusReport = {
+        ...report,
+        trap_cache_upload_duration_ms: Math.round(trapCacheUploadTime || 0),
+        trap_cache_upload_size_bytes: Math.round(
+          await getTotalCacheSize(config.trapCaches, logger),
```

Comment:
Only if also adding above.

```suggestion
          await getTotalCacheSize(config?.trapCaches || {}, logger),
```